FirebaseAdmin/FirebaseAdmin.Tests/FirebaseAppTest.cs

```diff
@@ -102,6 +104,28 @@ public void CreateAppOptions()
             Assert.Same(credential, copy.Credential);
             Assert.Equal("test-project", copy.ProjectId);
             Assert.Equal("test@service.account", copy.ServiceAccountId);
+            Assert.Equal(typeof(MockHttpClientFactory), copy.HttpClientFactory.GetType());
+        }
+
+        [Fact]
+        public void CreateAppOptionsNoClientFactory()
+        {
+            var credential = GoogleCredential.FromAccessToken("token");
+            var options = new AppOptions()
+            {
+                Credential = credential,
+                ProjectId = "test-project",
+                ServiceAccountId = "test@service.account",
+            };
+            var app = FirebaseApp.Create(options);
+            Assert.Equal("[DEFAULT]", app.Name);
+
+            var copy = app.Options;
+            Assert.NotSame(options, copy);
+            Assert.Same(credential, copy.Credential);
+            Assert.Equal("test-project", copy.ProjectId);
+            Assert.Equal("test@service.account", copy.ServiceAccountId);
+            Assert.Equal(typeof(HttpClientFactory), copy.HttpClientFactory.GetType());

```

Comment:
Perhaps a not null check is enough here?

Also add a test case to verify that FirebaseApp doesn't allow null values for HttpClientFactory.